File: _scratch/run/make_the_string_great/sol.py
class Solution:
    def makeGood(self, s):
        st = []
        for c in s:
            if st and st[-1] != c and st[-1].lower() == c.lower():
                st.pop()
            else:
                st.append(c)
        return "".join(st)
import sys
_l=sys.stdin.read().split('\n')
_T=int(_l[0]);_o=[]
for _i in range(_T):
    _s=_l[1+_i] if 1+_i<len(_l) else ""
    _o.append(str(Solution().makeGood(_s)))
print('\n'.join(_o))

File: _scratch/run/make_the_string_great/sol.js
/**
 * @param {string} s
 * @return {string}
 */
var makeGood = function(s) {
    const st = [];
    for (const c of s) {
        if (st.length && st[st.length - 1] !== c && st[st.length - 1].toLowerCase() === c.toLowerCase()) st.pop();
        else st.push(c);
    }
    return st.join('');
};
const _l=require('fs').readFileSync(0,'utf8').split('\n');const _T=+_l[0];const _o=[];
for(let _i=0;_i<_T;_i++){const _s=(_l[1+_i]||'').replace(/\r$/,'');_o.push(String(makeGood(_s)));}
console.log(_o.join('\n'));